tests/test_all_integration.py

```diff
@@ -339,26 +351,26 @@ def test_communication_manager_integration(
         time.sleep(0.1)
     assert len(scheduler.processed) == 5, "Averaging failed/not completed in time!"
     # we've only heard one set of new weights so our listen_iters are 1
-    assert communication_manager.optimizer.curr_averages_this_round == 1, \
-        "Should have only listened once!"
-    # now the communication manager should be idle
-    scheduler.runners_run_next_jobs()
-    time.sleep(0.1)
-    assert len(scheduler.processed) == 5, "No job should have been run!"
-    # now it should hear more new weights
-    communication_manager.inform(
-        RawEventTypes.NEW_MESSAGE.name,
-        new_weights_event
-    )
-    assert communication_manager.optimizer.job.job_type == JobTypes.JOB_AVG.name, \
-        "Should be ready to average!"
-    timeout = time.time() + 3
-    while time.time() < timeout and len(scheduler.processed) == 5:
-        # second averaging job
-        scheduler.runners_run_next_jobs()
-        time.sleep(0.1)
-    assert len(scheduler.processed) == 6, "Averaging failed/not completed in time!"
-    # we've heard both sets of new weights so our listen_iters are 2
+    # assert communication_manager.optimizer.curr_averages_this_round == 1, \
+    #     "Should have only listened once!"
+    # # now the communication manager should be idle
+    # scheduler.runners_run_next_jobs()
+    # time.sleep(0.1)
+    # assert len(scheduler.processed) == 5, "No job should have been run!"
+    # # now it should hear more new weights
+    # communication_manager.inform(
+    #     RawEventTypes.NEW_MESSAGE.name,
+    #     new_weights_event
+    # )
+    # assert communication_manager.optimizer.job.job_type == JobTypes.JOB_AVG.name, \
+    #     "Should be ready to average!"
+    # timeout = time.time() + 3
+    # while time.time() < timeout and len(scheduler.processed) == 5:
+    #     # second averaging job
+    #     scheduler.runners_run_next_jobs()
+    #     time.sleep(0.1)
+    # assert len(scheduler.processed) == 6, "Averaging failed/not completed in time!"
+    # # we've heard both sets of new weights so our listen_iters are 2
```

Comment:
Did you intend to leave this commented out?

Reply:
Yep. This indicates the portion of the communication manager test that has been removed because of the state-setting feature, which I think shows the consequence on the actual algorithm the best.

Reply:
I think you should leave a note of that right before the commented block or honestly just remove it (I would still see the removal either way, but this way the intended impact of the design decision is clearer here)

Reply:
Was planning to remove this upon approval of the PR like the other test I mentioned in the opening note of the PR anyways, yep